arch/xtensa/include/esp32/memory_layout.h

```diff
@@ -32,66 +32,121 @@
  *
  * CONFIG_HEAP2_BASE                         eg. 3f80 0000
  *     :
- *     : g_mmheap region3 (CONFIG_ESP32_SPIRAM)
+ *     : g_mmheap (CONFIG_ESP32_SPIRAM)
  *     :
  * CONFIG_HEAP2_BASE + CONFIG_HEAP2_SIZE     eg. 3fc0 0000
  *
- * _sheap                                    eg. 3ffc 8c6c
+ * HEAP_REGION0_START                        3ffa e6f0
  *     :
- *     : g_iheap (CONFIG_XTENSA_IMEM_USE_SEPARATE_HEAP)
+ *     : g_mmheap region0
  *     :
- * _sheap + CONFIG_XTENSA_IMEM_REGION_SIZE   eg. 3ffd ebfc
+ * HEAP_REGION0_END                          3ffa fff0
+ *     :
+ * _sheap                                    eg. 3ffc 8c6c
  *     :
  *     : g_mmheap region1
  *     :
  * HEAP_REGION1_END                              3ffd fff0
  *     :
  *     : ROM data
  *     :
- * HEAP_REGION2_START                            3ffe 1330 or 3ffe 7e40
+ * HEAP_REGION2_START                            3ffe 0450
+ *     :
+ *     : g_iheap (CONFIG_XTENSA_IMEM_USE_SEPARATE_HEAP)
```

Comment:
this iheap region is only for !SMP, right?

i guess it's more straightforward to make the `if CONFIG_SMP` block below
start from `HEAP_REGION2_START`.
that is:
```
------------
if !CONFIG_SMP

HEAP_REGION2_START
:
(omit)
:
------------
if CONFIG_SMP

HEAP_REGION2_START
:
(omit)
:
------------
_eheap
```
how do you think?

Reply:
I did so, thanks for the suggestion.